chiselFrontend/src/main/scala/chisel3/internal/Error.scala

```diff
@@ -38,12 +46,8 @@ private[chisel3] class ErrorLog {
     val depTag = s"[${Console.BLUE}deprecated${Console.RESET}]"
     val warnTag = s"[${Console.YELLOW}warn${Console.RESET}]"
     val errTag = s"[${Console.RED}error${Console.RESET}]"
-    deprecations foreach { case ((message, stackElement), count) =>
-      val line = stackElement match {
-        case Some(stackElement) => s"$depTag ${stackElement.getFileName}:${stackElement.getLineNumber} ($count calls): $message"
-        case None => s"$depTag (unknown location) ($count calls): $message"
-      }
-      println(line)
+    deprecations foreach { case ((message, sourceLoc), count) =>
```

Comment:
Since foreach is not purely functional it should be called via dot notation

Reply:
Fixed